ctapipe/reco/FitGammaHillas.py

```diff
@@ -0,0 +1,308 @@
+                                )
+            circle.weight = moments.size * (moments.length/moments.width)
+            self.circles[tel_id] = circle
+            
+    
+    def fit_origin_crosses(self):
+        """ calculates the origin of the gamma as the weighted average direction
+            of the intersections of all great circles
+        """
+        
+        assert len(self.circles) >= 2, "need at least two telescopes, have {}".format(len(self.circles))
+        
+        crossings = []
+        for perm in combinations(self.circles.values(), 2):
+            n1,n2 = perm[0].norm, perm[1].norm
+            # cross product automatically weighs in the angle between the two vectors
+            # narrower angles have less impact, perpendicular vectors have the most
+            crossing = np.cross(n1,n2)
+            # two great circles cross each other twice
+            # (one would be the origin, the other one the direction of the gamma)
+            # it doesn't matter which we pick but it should at least be consistent
+            # make sure to always take the "upper" solution
+            if crossing[2] < 0: crossing *= -1
+            crossings.append( crossing*perm[0].weight * perm[0].weight  )
+        # averaging over the solutions of all permutations
+        return linalg.normalise(sum(crossings))*u.dimless, crossings
+            
+            
+
+    def fit_origin_minimise(self, seed=[0,0,1], test_function=None):
+        """ fits the origin of the gamma with a minimisation procedure
+            this function expects that get_great_circles has been run already
+            a seed should be given otherwise it defaults to "straight up"
+            supperted functions to minimise are an M-estimator and the 
+            negative sum of the angles to all normal vectors of the 
+            great circles 
+            
+            Parameters:
+            -----------
+            seed : length-3 array
+                starting point of the minimisation
+            test_function : member function if this class
+                either _n_angle_sum or _MEst (or own implementation...)
+                defaults to _n_angle_sum if none is given
+                _n_angle_sum seemingly superior to _MEst
+            
+            Returns:
+            --------
+            direction : length-3 numpy array as dimensionless quantity
+                best fit for the origin of the gamma from the minimisation process
+        """
+        
+        if test_function == None: test_function = self._n_angle_sum
+        '''
+            using the sum of the cosines of each direction with every other direction
+            don't use the product -- with many steep angles, the product will 
+            become too small and the weight (and the whole fit) useless
+        '''
+        weights = [ np.sum( [ linalg.length( np.cross(A.norm,B.norm) ) for A in self.circles.values() ] ) *B.weight 
```

Comment:
in general , run `pylint` on the code if your editor doesn't do auto PEP8 compliance (or run `autopep8`, which you can install via `pip install autopep8` or `conda install -c conda-forge autopep8`